src/gazebo_ros_pulse_lidar.cpp

```diff
@@ -33,22 +47,23 @@ namespace gazebo
      // Safety check
      if (_model->GetJointCount() == 0)
      {
-       std::cerr << "Invalid joint count, NPS Gazebo ROS lidar plugin not loaded\n";
+       std::cerr << "Invalid joint count; "
+                 << "NPS Gazebo ROS lidar plugin not loaded." << std::endl;
        return;
```

Comment:
Do not use cerr and cout streams.  Use the ROS logging system.  We use logging instead of cerr because everything is threaded, which obfuscates where errors come from.  If fatal use "ROS_FATAL_STREAM_NAMED("gazebo_ros_pulse_lidar", "Invalid joint count.".  If info use ROS_INFO_NAMED.

Here it looks like two errors got mixed and the check for "plugin not loaded" is missing.